tests/unit/store/asserts-test.js

```diff
@@ -0,0 +1,36 @@
+import {createStore} from 'dummy/tests/helpers/store';
+
+import {module, test} from 'qunit';
+
+module("unit/store/asserts - DS.Store methods produce useful assertion messages");
+
+const MODEL_NAME_METHODS = [
+  'createRecord',
+  'findRecord',
+  'findByIds',
+  'peekRecord',
+  'hasRecordForId',
+  'recordForId',
+  'query',
+  'queryRecord',
+  'findAll',
+  'peekAll',
+  'filter',
+  'recordIsLoaded',
+  'modelFor',
+  'modelFactoryFor',
```

Comment:
I think its ok to leave the asserts on private APIs. If these methods were ever marked public in the future its unlikely they someone would remember to come back and update this test.